src/Makefile.am

```diff
@@ -638,6 +639,7 @@ libbitcoin_common_a_SOURCES = \
   compressor.cpp \
   core_read.cpp \
   core_write.cpp \
+  cs_main.cpp \
```

Comment:
Why is this in common when it should be in kernel?

Reply:
Right. Moved to util.

Reply:
Why is this in util when it should be in kernel?

Reply:
It also needs to be in common or util, otherwise you wont be able to link without the kernel?

Reply:
Yes, bitcoin_kernel is linked into bitcoin_node according to https://github.com/bitcoin/bitcoin/blob/master/doc/design/libraries.md#dependencies

cs_main shouldn't be put into any other binary or library

Reply:
Right. I didn't understand you actually meant move this to `kernel/cs_main.*`, and have that exist in the node lib. I have moved things around now.